mps-model-adapters/src/main/kotlin/org/modelix/model/mpsadapters/MPSNode.kt

```diff
@@ -145,10 +145,41 @@ data class MPSNode(val node: SNode) : IDefaultNodeAdapter {
 
     override fun addNewChild(role: IChildLink, index: Int, concept: IConceptReference?): INode {
         val repo = checkNotNull(node.model?.repository)
-        val targetConcept = concept?.let { MPSLanguageRepository(repo).resolveConcept(it.getUID()) }
+        val targetConcept = concept?.let {
+            MPSLanguageRepository(repo).resolveConcept(it.getUID())
+                ?: MPSConcept.tryParseUID(it.getUID())
+        }
+
+        // A null value for the concept would default to BaseConcept, but then BaseConcept should be used explicitly.
+        checkNotNull(targetConcept) { "MPS concept not found: $concept" }
+
         return addNewChild(role, index, targetConcept)
     }
 
+    override fun addNewChildren(role: String?, index: Int, concepts: List<IConceptReference?>): List<INode> {
+        requireNotNull(role) { "containment link required" }
+        val link = MPSChildLink(getMPSContainmentLink(role))
+        return addNewChildren(link, index, concepts)
+    }
+
+    override fun addNewChildren(link: IChildLink, index: Int, concepts: List<IConceptReference?>): List<INode> {
+        return concepts.asReversed().map { addNewChild(link, index, it) }.asReversed()
```

Comment:
Why do we need `asReversed()` twice here?

Reply:
Because they are inserted using the same index and would then end up in reversed order in the model, but that's actually not true if -1 is used for the index, which would append them in the correct order. I fixed that. Should also be easier to understand now.